agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4866,6 +4867,23 @@ bool slave_thread::handle_client_steering_request(Socket *sd, ieee1905_1::CmduMe
 
     //check if the Request Mode bit is set
     if (request_mode) {
+        //TODO Handle 0 or more then 1 sta in list, currenlty cli steers only 1 client
+        auto request_out = message_com::create_vs_message<
+            beerocks_message::cACTION_APMANAGER_CLIENT_BSS_STEER_REQUEST>(cmdu_tx, mid);
+        if (request_out == nullptr) {
+            LOG(ERROR) << "Failed building ACTION_APMANAGER_CLIENT_BSS_STEER_REQUEST message!";
+            return false;
+        }
+        auto bssid_list = steering_request_tlv->target_bssid_list(0);
+
+        request_out->params().mac            = std::get<1>(steering_request_tlv->sta_list(0));
+        request_out->params().disassoc_timer = steering_request_tlv->btm_disassociation_timer();
+        request_out->params().bssid.mac      = std::get<1>(bssid_list).target_bssid;
+        request_out->params().bssid.channel  = std::get<1>(bssid_list).target_bss_channel_number;
+        request_out->params().disassoc_imminent =
+            steering_request_tlv->request_flags().btm_disassociation_imminent_bit;
+
+        message_com::send_cmdu(ap_manager_socket, cmdu_tx);
```

Comment:
From the standard:

> If a Multi-AP Agent receives a Client Steering Request message with Request Mode bit set to one, then it shall respond within one second with a 1905 Ack message. If a STA specified in the Client Steering Request message is not associated with the source BSSID specified in the same message (an error scenario), for each of those unassociated STAs, the Multi-AP Agent shall include an Error Code TLV with the reason code field set to 0x02 and the STA MAC address field included per section 17.2.36 in the 1905 Ack message.

You will have to query the AP manager if the station is connected, and return 1905 ACK message accordingly.
So what @CoralMalachi did in lines 4860-4864 needs to be replaced.

Reply:
Moved to handle in manager.